.github/workflows/scripts/bump-versions.sh

```diff
@@ -0,0 +1,210 @@
+    *) echo "Unknown parameter passed: $1"; usage ;;
+  esac
+  shift
+done
+
+# Check if --pull-request is passed without --commit
+if [ "$create_pr" = true ] && [ "$commit_changes" = false ]; then
+  echo "--pull-request requires --commit"
+  usage
+fi
+
+# Check if at least one of the required arguments is provided
+if [ -z "$current_beta_core" ] && [ -z "$current_beta_contrib" ] && [ -z "$current_stable" ]; then
+  usage
+fi
+
+# Validate and strip versions
+if [ -n "$current_beta_core" ]; then
+  validate_and_strip_version current_beta_core
+fi
+if [ -n "$current_beta_contrib" ]; then
+  validate_and_strip_version current_beta_contrib
+fi
+if [ -n "$current_stable" ]; then
+  validate_and_strip_version current_stable
+fi
+if [ -n "$next_beta_core" ]; then
+  validate_and_strip_version next_beta_core
+fi
+if [ -n "$next_beta_contrib" ]; then
+  validate_and_strip_version next_beta_contrib
+fi
+if [ -n "$next_stable" ]; then
+  validate_and_strip_version next_stable
+fi
+
+# Function to bump the minor version and reset patch version to 0
+bump_version() {
+  local version=$1
+  local major
+  major=$(echo "$version" | cut -d. -f1)
+  local minor
+  minor=$(echo "$version" | cut -d. -f2)
+  local new_minor
+  new_minor=$((minor + 1))
+  echo "$major.$new_minor.0"
+}
+
+# Infer the next beta version if not supplied
+if  [ -n "$current_beta_core" ] && [ -z "$next_beta_core" ]; then
+  next_beta_core=$(bump_version "$current_beta_core")
+fi
+if  [ -n "$current_beta_contrib" ] && [ -z "$next_beta_contrib" ]; then
+  next_beta_contrib=$(bump_version "$current_beta_contrib")
+fi
+
+# Infer the next stable version if current_stable provided and next version not supplied
+if [ -n "$current_stable" ] && [ -z "$next_stable" ]; then
+  next_stable=$(bump_version "$current_stable")
+fi
+
+# List of files to update
+files=(
+  "distributions/otelcol-contrib/manifest.yaml"
+  "distributions/otelcol/manifest.yaml"
+  "distributions/otelcol-k8s/manifest.yaml"
+  "distributions/otelcol-otlp/manifest.yaml"
+  "Makefile"
+)
+
+# Update versions in each file
+for file in "${files[@]}"; do
+  if [ -f "$file" ]; then
+    temp_file=$(mktemp)
+    cp "$file" "$temp_file"
+    line_number=1
+    while IFS= read -r line; do
+      if [[ "$line" == *"github.com/open-telemetry/opentelemetry-collector-contrib"* ]]; then
+        if [ -n "$current_beta_contrib" ]; then
+          sed -i.bak "${line_number}s/$current_beta_contrib/$next_beta_contrib/" "$temp_file"
+        fi   
+      elif [[ "$line" == *"go.opentelemetry.io/collector"* ]]; then
+        if [ -n "$current_beta_core" ]; then
+          sed -i.bak "${line_number}s/$current_beta_core/$next_beta_core/" "$temp_file"
+        fi
+        if [ -n "$current_stable" ]; then
+          sed -i.bak "${line_number}s/$current_stable/$next_stable/" "$temp_file"
+        fi
+      fi
+      line_number=$((line_number + 1))
+    done < "$file"
+    mv "$temp_file" "$file"
+    rm "${temp_file}.bak"
```

Comment:
The script logs errors trying to remove non-existent `.bak` files when run with only the `--current-beta-contrib` flag:

```console
./.github/workflows/scripts/bump-versions.sh --current-beta-contrib v0.111.0
rm: cannot remove '/tmp/tmp.h6AQVOKLTG.bak': No such file or directory
rm: cannot remove '/tmp/tmp.VsQcTga32N.bak': No such file or directory
Version update completed.
Changes not committed and PR not created.
```

I propose to add an existence check before removing the `.bak` files.

```suggestion
    [ -f "${temp_file}.bak" ] && rm "${temp_file}.bak"
```

Reply:
got rid of the backup logic so this problem should be solved